test/index.js

```diff
@@ -78,10 +79,17 @@ describe("Test for infinity menu", function() {
 			onLeafMouseUp={onLeafMouseUpStub}
 			onLeafMouseDown={onLeafMouseDownStub}
 			onClose={onCloseStub}
+			headerMetadata={headerMetadata}
 			/>;
 		dom = TestUtils.renderIntoDocument(component);
 	});
 
+	it("should render correctly with header metadata", function () {
```

Comment:
while this test passes it's not really testing anything. It's barely more than saying `true === true` , you're explicitly passing an argument to a function and then asserting that you passed that argument. This test would pass whether or not `headerMetadata was a prop that you passed through to the Header. to me this is a much more valid test

```js
	it("should pass arbitrary props through to a custom header component", function() {
		class Header extends React.Component {
			render() {
				return (
					<div>{this.props.randomProp}</div>
				);
			}
		}
		component = <InfinityMenu
			tree={tree}
			onNodeMouseClick={onNodeMouseClickStub}
			onLeafMouseClick={onLeafMouseClickStub}
			onLeafMouseUp={onLeafMouseUpStub}
			onLeafMouseDown={onLeafMouseDownStub}
			onClose={onCloseStub}
			headerContent={Header}
			headerMetadata={{
				randomProp: "hello"
			}}
		/>;
		shallowRenderer.render(component);
		const result = shallowRenderer.getRenderOutput();
		assert.equal(result.props.children[0].props.randomProp, "hello");
	});
```